Added test for ParseInterpolatedNoEnvVarsTest

WhiteOlivierus opened the pull request:
I would expect this to work like `ParseInterpolatedTest` but it does not replace the enviroment variables correctly. Now it does.

Olivierus committed:
Added failing test

WhiteOlivierus committed:
Fixed the bug

WhiteOlivierus committed:
All the tests are fixed now

rogusdev commented:
I will look this over more closely, but a couple of things right off:
1. You changed a lot of whitespace, including in test assertions, which is definitely not appropriate to change.
2. What is the actual test that was failing? I am not following what specifically you found to be "not working". Please elaborate.

WhiteOlivierus commented:
Oh yes the whitespace did change, I will resolve that.

I have added ParseInterpolatedNoEnvTest as a test because I want to only load the environment variables as a dictionary. But when I did the values in my env file like TEST2=${TEST} resolved to empty string. So I fixed that locally, but the pipeline gives different results from local.

rogusdev commented:
Until you fix the whitespace changes, I will not review this.

Also, is this the same thing as https://github.com/tonerdo/dotnet-env/issues/92 ?

WhiteOlivierus commented:
I will fix the whitespace next week, because I am on vacation right now.

And yes it seems that it is the same problem. I want to resolve all the $env without loading them in the environment. I just want them returned as a dictionary.

WhiteOlivierus committed:
Fixed some spacing

WhiteOlivierus committed:
spaces instead of tabs

WhiteOlivierus committed:
spaces instead of tabs

WhiteOlivierus commented:
@rogusdev I fixed the whitespace issues

rogusdev commented:
Your PR actually sets env vars, which is exactly what it should not do (even only temporarily). I created a solution that keeps all such vars for no env var interpolation in memory: https://github.com/tonerdo/dotnet-env/pull/97 How does this work for you?

rogusdev commented:
Also, you changed whitespace in tests, so your tests are failing in CI. But focus on whether or not you are ok with my alternative PR, or if you have any comments for that approach. That's what I am going to merge before your current implementation.

WhiteOlivierus commented:
@rogusdev I have read through it and if it works and the test does complete this is exactly what I was looking for.
